src/linear_algebra/quadratic_form.lean

```diff
@@ -139,21 +199,80 @@ variables {N : Type v} [add_comm_group N] [module R N]
 def comp (Q : quadratic_form R N) (f : M →ₗ[R] N) :
   quadratic_form R M :=
 { to_fun := λ x, Q (f x),
-  to_fun_smul := λ a x, by simp [map_smul],
-  polar_add_left := λ x x' y,
-    by convert Q.polar_add_left (f x) (f x') (f y) using 1; simp [polar],
-  polar_smul_left := λ a x y,
-    by convert Q.polar_smul_left a (f x) (f y) using 1; simp [polar],
-  polar_add_right := λ x y y',
-    by convert Q.polar_add_right (f x) (f y) (f y') using 1; simp [polar],
-  polar_smul_right := λ a x y,
-    by convert Q.polar_smul_right a (f x) (f y) using 1; simp [polar] }
+  to_fun_smul := λ a x, by simp only [map_smul, f.map_smul],
+  polar_add_left' := λ x x' y,
+    by convert polar_add_left (f x) (f x') (f y) using 1;
+      simp only [polar, f.map_add],
+  polar_smul_left' := λ a x y,
+    by convert polar_smul_left a (f x) (f y) using 1;
+      simp only [polar, f.map_smul, f.map_add, smul_eq_mul],
+  polar_add_right' := λ x y y',
+    by convert polar_add_right (f x) (f y) (f y') using 1;
+      simp only [polar, f.map_add],
+  polar_smul_right' := λ a x y,
+    by convert polar_smul_right a (f x) (f y) using 1;
+      simp only [polar, f.map_smul, f.map_add, smul_eq_mul] }
 
 @[simp] lemma comp_apply (Q : quadratic_form R N) (f : M →ₗ[R] N) (x : M) :
   (Q.comp f) x = Q (f x) := rfl
 
 end comp
 
+section comm_ring
+
+/-- Create a quadratic form in a commutative ring by proving only one side of the bilinearity. -/
+def mk_left (f : M → R₁)
+  (to_fun_smul : ∀ a x, f (a • x) = a * a * f x)
+  (polar_add_left : ∀ x x' y, polar f (x + x') y = polar f x y + polar f x' y)
+  (polar_smul_left : ∀ a x y, polar f (a • x) y = a * polar f x y) :
+  quadratic_form R₁ M :=
+{ to_fun := f,
+  to_fun_smul := to_fun_smul,
+  polar_add_left' := polar_add_left,
+  polar_smul_left' := polar_smul_left,
+  polar_add_right' :=
+    λ x y y', by rw [polar_comm, polar_add_left, polar_comm f y x, polar_comm f y' x],
+  polar_smul_right' :=
+    λ a x y, by rw [polar_comm, polar_smul_left, polar_comm f y x, smul_eq_mul] }
+
+/-- The product of linear forms is a quadratic form. -/
+def lin_mul_lin (f g : M →ₗ[R₁] R₁) : quadratic_form R₁ M :=
+mk_left (f * g)
+  (λ a x, by { simp, ring } )
+  (λ x x' y, by { simp [polar], ring })
+  (λ a x y, by { simp [polar], ring })
+
+@[simp]
+lemma lin_mul_lin_map (f g : M →ₗ[R₁] R₁) (x) : lin_mul_lin f g x = f x * g x := rfl
```

Comment:
```suggestion
lemma lin_mul_lin_apply (f g : M →ₗ[R₁] R₁) (x) : lin_mul_lin f g x = f x * g x := rfl
```

Reply:
It looks like some other definitions in this file also use `_map`. Should I change those as well?